src/rx-query.ts

```diff
@@ -686,9 +797,82 @@ function __ensureEqual<RxDocType>(rxQuery: RxQueryBase<RxDocType>): Promise<bool
                     rxQuery._setResultData(newResultData as any);
                 }
                 return ret;
+            })
+            .then(async (returnValue) => {
+                if (shouldUpdatePersistentQueryCache(rxQuery)) {
+                  await updatePersistentQueryCache(rxQuery);
+                }
+                return returnValue;
             });
     }
-    return Promise.resolve(ret); // true if results have changed
+
+    return ret; // true if results have changed
+}
+
+function shouldUpdatePersistentQueryCache<RxDocType>(rxQuery: RxQueryBase<RxDocType>) {
+    if (!rxQuery._persistentQueryCacheBackend) {
+        return false;
+    }
+
+    if (!rxQuery._persistentQueryCacheResult) {
+        return true;
+    }
+
+    if (!rxQuery._persistentQueryCacheResultLwt) {
+        return true;
+    }
+
+    const lwt = Math.floor(Number(rxQuery._persistentQueryCacheResultLwt));
+    const ttl = 3600 * 1000; // 1hr
```

Comment:
ooc what's the rationale behind this?

Reply:
If the query cache was persisted more than an hour ago, and either event-reduce or exec of the query runs, it persists the current query cache. This eventually refreshes the cache, but at a very slow pace. I don't know what a good TTL is for this, I just picked an arbitrary value. Could be a day as well, or 1min.